extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/NativeImageProcessor.java

```diff
@@ -68,6 +71,66 @@
+                "org.apache.camel.component.netty.NettyServerBootstrapConfiguration",
+                "org.apache.camel.component.fhir.FhirUpdateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirOperationEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirConfiguration",
+                "org.apache.camel.component.fhir.FhirLoadPageEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirSearchEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirTransactionEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirCreateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirValidateEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirReadEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirCapabilitiesEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirHistoryEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirMetaEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirPatchEndpointConfiguration",
+                "org.apache.camel.component.fhir.FhirDeleteEndpointConfiguration",
+
+                /* org.apache.camel.component.consul.* can be removed after the upgrade to Camel 3.1 */
+                "org.apache.camel.component.consul.ConsulConfiguration",
+                "org.apache.camel.component.consul.ConsulClientConfiguration",
+                "org.apache.camel.component.consul.health.ConsulHealthCheckRepositoryConfiguration",
+                "org.apache.camel.component.consul.cloud.ConsulServiceRegistryConfiguration"));
+
+        @BuildStep
+        void bannedReflectiveClasses(
+                CombinedIndexBuildItem combinedIndex,
+                List<ReflectiveClassBuildItem> reflectiveClass,
+                BuildProducer<GeneratedResourceBuildItem> dummy // to force the execution of this method
+        ) {
+            final DotName uriParamsDotName = DotName.createSimple("org.apache.camel.spi.UriParams");
+
+            final Set<String> bannedClassNames = combinedIndex.getIndex()
+                    .getAnnotations(uriParamsDotName)
+                    .stream()
+                    .filter(ai -> ai.target().kind() == Kind.CLASS)
+                    .map(ai -> ai.target().asClass().name().toString())
+                    .collect(Collectors.toSet());
+
+            Set<String> violations = reflectiveClass.stream()
+                    .map(ReflectiveClassBuildItem::getClassNames)
+                    .flatMap(Collection::stream)
+                    .filter(cl -> !URI_PARAMS_WHITELIST.contains(cl))
+                    .filter(bannedClassNames::contains)
+                    .collect(Collectors.toSet());
+
+            if (!violations.isEmpty()) {
+                throw new IllegalStateException(
```

Comment:
all the camel components have a property that can control if the generated configurer have to be used or not so I feel we should log a warning but I'd rather avoid to fail the build.